cupy/cuda/compiler.py

```diff
@@ -276,18 +276,26 @@ def get_cache_dir():
 _empty_file_preprocess_cache = {}
 
 
-def compile_with_cache(source, options=(), arch=None, cache_dir=None,
-                       extra_source=None, backend='nvrtc'):
+def compile_with_cache(
+        source, options=(), arch=None, cache_dir=None, extra_source=None,
+        backend='nvrtc', grid_sync=False):
```

Comment:
Isn't this function public? It may be better to make the argument keyword-only.

Reply:
Fixed.